src/traces/heatmap/plot.js

```diff
@@ -109,11 +153,18 @@ module.exports = function(gd, plotinfo, cdheatmaps, heatmapLayer) {
             y = cd0.yfill;
         }
 
+        var drawingMethod = 'default';
+        if(zsmooth) {
+            drawingMethod = zsmooth === 'best' ? 'smooth' : 'fast';
+        } else if(trace._islinear && xGap === 0 && yGap === 0 && supportsPixelatedImage()) {
```

Comment:
Could we store the result of `supportsPixelatedImage` function in a variable and return that the second time it called.

Reply:
This has already been addressed. Resolving...